### ISSUE TITLE
Update curl from 8.4.0 to 8.6.0 due to security vulnerabilities CVE-2023-46219 and CVE-2023-46218

### ISSUE THREAD

**Issue by jsasswis** [tensorflow/tensorflow#69799]
### Issue type

Build/Install

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

binary

### TensorFlow version

2.15.1, 2.16.1

### Custom code

No

### OS platform and distribution

Ubuntu 22.04

### Python version

3.9, 3.10, 3.11

### Current behavior?

Currently, TensorFlow uses curl 8.4.0 which has been identified with security vulnerabilities CVE-2023-46219 and CVE-2023-46218. Not affected versions include curl >=8.5.0

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/workspace2.bzl#L423

Info on CVEs identified in curl 8.4.0:
https://nvd.nist.gov/vuln/detail/CVE-2023-46219
https://nvd.nist.gov/vuln/detail/CVE-2023-46218

Curl fix documentation:
https://curl.se/docs/CVE-2023-46219.html
https://curl.se/docs/CVE-2023-46218.html

Is there any reason for TensorFlow not to update curl? If no, I'm willing to submit a PR to address this.

### Standalone code to reproduce the issue

```shell
N/A as this is a request for a third party update during build based on security advisories.
```

### Relevant log output

```shell
N/A
```

**Comment by tilakrayal** [comment]
@jsasswis,
Could you please confirm whether the latest tensorflow build is compatible with the above mentioned versions. If yes, please feel free to contribute with the PR. Thank you

**Comment by mihaimaruseac** [comment]
@rtg0795 as this should be done before final releases

**Comment by mihaimaruseac** [comment]
Would get fixed by #70117
